# 用户自定义异常类

- 一般地，用户自定义异常类都是RuntimeException的子类

- 自定义异常类通常需要编写几个重载的构造器

- 自定义异常需要提供serialVersionUID

- 自定义的异常通过throw抛出

- 自定义异常最重要的是异常类的名字，当异常出现时，可以根据名字判断异常类型

在Java中，我们可以通过扩展 `Exception` 类或其子类来定义用户自定义异常类。下面是一个示例，展示了

如何定义一个自定义异常类：

```java
public class CustomException extends Exception {

  public CustomException(String message) {
    super(message);
  }

  public CustomException(String message, Throwable cause) {
    super(message, cause);
  }
  
  // 自定义异常类的其他方法和属性
}
```

在这个示例中，我们创建了一个名为 `CustomException` 的自定义异常类，它扩展了 `Exception` 类。我们定义了两个构造函数，分别用于初始化异常消息和异常原因。我们还可以在这个类中添加其他方法和属性，以便在抛出异常时提供更多信息。

使用这个自定义异常类，我们可以在我们的程序中抛出它，例如：

```java
public class Example {
  public static void main(String[] args) throws CustomException {
    // 在某些情况下抛出自定义异常
    throw new CustomException("Something went wrong");
  }
}
```

在这个示例中，我们在 `main` 方法中抛出了一个自定义异常 `CustomException`。当这个异常被抛出时，它将包含一条消息 "Something went wrong"，并且程序将停止运行。我们还可以通过 `try-catch` 块来捕获和处理这个异常。

总之，我们可以通过扩展 `Exception` 类或其子类来定义用户自定义异常类，并在程序中抛出它来表示特定的异常情况。通过定义自己的异常类，我们可以更好地管理我们的程序中的异常情况，并提供更多信息以便于调试和修复问题。
